.github/workflows/build-luadocs.yml

```
@@ -0,0 +1,56 @@
+name: Build Pioneer Lua Documentation
+
+on:
+  push:
+    branches:
+    - master
+    paths:
+    - 'src/lua/**.cpp'
+    - 'data/libs/**.lua'
+    - 'data/pigui/**.lua'
+    
+jobs:
+  build-lua-docs:
+    runs-on: ubuntu-20.04
+    
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v2
+      with:
+        ref: master
+        fetch-depth: 0
+        
+    - name: Setup git identity
+      run: |
+        git config --local user.name "Commander Jameson"
+        git config --local user.email "CmdrJameson@pioneerspacesim.net"
+        
+    - name: Merge master into gh-pages
+      run: |
+        git checkout gh-pages
+        git pull --no-rebase origin master
```

Adding on to what I suggested in IRC, this is what the action would look like using a bare branch - very little would need to change, other than the gh-pages config serving out of the root of the `gh-pages` branch instead of the `docs` subfolder.

```suggestion
    - name: Checkout gh-pages branch
      uses: actions/checkout@v2
      with:
        ref: gh-pages
        path: docs
        token: "CMDR Jameson's Token"
```